Assets/_BForBoss/_Core/Scripts/Enemies/IsLineOfSightBlocked.cs

```diff
@@ -0,0 +1,31 @@
+using System;
+using Perigon.Utility;
+using UnityEngine;
+
+namespace BForBoss
+{
+    public class IsLineOfSightBlocked
```

Comment:
Don't think this is necessary. Could probably make this a static helper class (i.e. `public static class LineOfSight`) and instead of `Execute` make it `public bool IsBlocked(Vector3 destination, Vector3 shootFromPosition){}`.

Passing this class as a variable seems like it adds no benefit atm since no information is being stored/kept.

Reply:
or since everything is now under the `BForBoss` namespace, you could make it `internal static class LineOfSight'

Reply:
It needs to store the `Func<Vector3>` not just Vector3. Every enemy has their own `Func<Vector3>` that constantly changes. 

If I were to make this a static class, then
1. bad for testing.
2. Then it'll be another callback from the shooting/navigation classes.

Reply:
I can change the name though

Reply:
Hey, sorry for the late reply. Honestly, I'm still not 100% convinced that this can't just be a helper class.

Like `EnemyShootingBehaviour` has both `getPlayerPosition` and `shootingFromPosition` so `LineOfSight' has no need for having its information.

And for `AgentNavigationBehaviour` has `navigationDesitnation`. So feasibly we could pass in Another Func<Vector3> (i.e. `getPlayerPosition`) into the Initialize Function instead of `LineOfSight` and instead Call the `LineOfSight` static helper class.

Let's talk about this during the Work session today!

Reply:
The reason of having a separate class handle it is 
1. reusability.
2. Doesn't break SOLID principle (single responsibility principle). The job of  floating target is for connecting everything, having a random helper function there would signal the fact that all shared logic should be placed in the parent class.

If we really want to break reason 1, sure since we only have one ai for now.

If we want to break 2, then we'll have start having callbacks for any shared code between components in their initializes as a pattern. Which we have been doing, but also had to start refactoring since it went too far.


I'll leave this decision up to you if I can't change your mind after this.

Reply:
And havin LineOfSight makes it explicit as well, so that we know we need this explicit logic as opposed to. A generic function to pass in that could be any logic.

It's like when setting up a class initialize, it forces another dev to use line of sight instead of passing in a random function.